gravitee-apim-console-webui/src/management/api/endpoints-v4/backend-services/endpoints-group/api-endpoints-group.component.ts

```diff
@@ -0,0 +1,220 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, Inject, OnInit, OnDestroy } from '@angular/core';
+import { StateService } from '@uirouter/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { catchError, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { EMPTY, Subject, Subscription } from 'rxjs';
+
+import { ApiV4, EndpointGroupV4 } from '../../../../../entities/management-api-v2';
+import { UIRouterState, UIRouterStateParams } from '../../../../../ajs-upgraded-providers';
+import { ApiV2Service } from '../../../../../services-ngx/api-v2.service';
+import { SnackBarService } from '../../../../../services-ngx/snack-bar.service';
+import { isUniqueAndDoesNotMatchDefaultValue } from '../../../../../shared/utils';
+import { GioPermissionService } from '../../../../../shared/components/gio-permission/gio-permission.service';
+import { ProxyConfiguration } from '../../../../../entities/proxy';
+
+@Component({
+  selector: 'api-endpoints-group',
+  template: require('./api-endpoints-group.component.html'),
+  styles: [require('./api-endpoints-group.component.scss')],
+})
+export class ApiEndpointsGroupComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+  private SUCCESSFUL_ENDPOINT_CONFIGURATION_SAVE_MESSAGE = 'Configuration successfully saved!';
+
+  public api: ApiV4;
+  public isReadOnly: boolean;
+  public generalForm: FormGroup;
+  public groupForm: FormGroup;
+
+  public initialGroupFormValue: unknown;
+
+  constructor(
+    @Inject(UIRouterStateParams) private readonly ajsStateParams,
+    @Inject(UIRouterState) private readonly ajsState: StateService,
+    private readonly apiService: ApiV2Service,
+    private readonly snackBarService: SnackBarService,
+    private readonly permissionService: GioPermissionService,
+  ) {}
+
+  public ngOnInit(): void {
+    const apiId = this.ajsStateParams.apiId;
+
+    this.apiService
+      .get(apiId)
+      .pipe(
+        tap((api: ApiV4) => this.initializeComponent(api)),
+        takeUntil(this.unsubscribe$),
+      )
+      .subscribe();
+  }
+
+  public ngOnDestroy(): void {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+
+  public onSubmit(): Subscription {
+    return this.apiService
+      .get(this.ajsStateParams.apiId)
+      .pipe(
+        switchMap((api: ApiV4) => this.updateApi(api)),
```

Comment:
You should use the returned API to update `this.api` (maybe call `initializeComponent`)